src/main/java/com/puppycrawl/tools/checkstyle/checks/OrderedPropertiesCheck.java

```diff
@@ -0,0 +1,202 @@
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.List;
+import java.util.Properties;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import com.puppycrawl.tools.checkstyle.StatelessCheck;
+import com.puppycrawl.tools.checkstyle.api.AbstractFileSetCheck;
+import com.puppycrawl.tools.checkstyle.api.CheckstyleException;
+import com.puppycrawl.tools.checkstyle.api.FileText;
+
+/**
+ * This implementation checks that keys of properties files are sorted.
+ * It is using the binary order from String.CASE_INSENSITIVE_ORDER.compare()
+ * The sort direction is ascending.
+ *
+ *   <p>
+ *       Example properties file:
+ *   </p>
+ *   <pre>
+ *       &#64;=64 (it is the 'at' sign)
+ *       A=65
+ *       a=97
+ *       &auml;=132 - violation
+ *       key=107 than nothing
+ *       key.sub=k is 107 and dot is 46
+ *   </pre>
+ *   <ul>
+ *       <li>
+ *           The &#64; sign is on position 64 and comes first.
+ *       </li>
+ *       <li>
+ *           The capital A is on 65 and the lowercase a is on position 97 on the ascii table.
+ *       </li>
+ *       <li>
+ *           On line 4 you have an a umlaut. This belongs to the extended ascii table and comes at
+ *           last position. This is the error here.
+ *       </li>
+ *       <li>
+ *           Key and subkey are in correct order here, because only keys are relevant.
+ *           Therefore on line 5 you have only "key" an nothing behind.
+ *           On line 6 you have "key." The dot is on position 46 which is higher than nothing.
+ *       </li>
+ *   </ul>
+ */
+@StatelessCheck
+public class OrderedPropertiesCheck extends AbstractFileSetCheck {
+
+    /**
+     * Localization key for check violation.
+     */
+    public static final String MSG_KEY = "properties.notSorted.property";
+    /**
+     * Localization key for IO exception occurred on file open.
+     */
+    public static final String UNABLE_TO_OPEN_FILE = "Unable to open file: ";
+    /**
+     * Pattern matching single space.
+     */
+    private static final Pattern SPACE_PATTERN = Pattern.compile(" ");
+
+    /**
+     * Construct the check with default values.
+     */
+    public OrderedPropertiesCheck() {
+        setFileExtensions("properties");
+    }
+
+    @Override
+    protected void processFiltered(File file, FileText fileText) throws CheckstyleException {
+        final SequencedProperties properties = new SequencedProperties();
+        try (InputStream inputStream = Files.newInputStream(file.toPath())) {
+            properties.load(inputStream);
+        }
+        catch (IOException ex) {
+            throw new CheckstyleException(UNABLE_TO_OPEN_FILE + file.getPath(), ex);
+        }
+
+        String previousProp = "";
+
+        final Enumeration<Object> keys = properties.keys();
```

Comment:
`properties.keys();` => `properties.keySet().iterator()`

Reply:
If I would do this, than the check will not work. Because keySet() creates a new Set by using new Hashtable.KeySet(). On a Hashtable is the order not guaranteed.

Reply:
Then you will have to suppress the inspection.

Reply:
suppressed with
@noinspection EnumerationCanBeIteration